src/chainstate/stacks/boot/pox.clar

```diff
@@ -153,8 +169,33 @@
             )
         ;; no state at all
         none
-    )
-)
+    ))
+
+(define-private (check-caller-allowed)
+    (or (is-eq tx-sender contract-caller)
```

Comment:
This first branch `(is-eq tx-sender contract-caller)` is only ever true if this method is being invoked by a contract-call transaction, right?

Reply:
Yep -- or is invoked by a contract in an `(as-contract ...)` closure.